Drivers/IWDG_driver/Src/IndependentWatchdog.cpp

```diff
@@ -0,0 +1,45 @@
+#include "IndependentWatchdog.h";
+#include "main.h"
+
+/**
+ * @brief Borrow the configuration of an already existing watchdog and set it
+ * to the prescaler and Reload values.
+ *
+ */
+IWDG_t::IWDG_t(IWDG_HandleTypeDef *watchdog) {
+	IWDG_InitTypeDef initDef = watchdog->Init;
+	this->iwdg_prescaler = initDef.Prescaler;
+	this->iwdg_reload = initDef.Reload;
+}
+
+/**
+ * @brief Initalize the driver using custom IWDG_Prescaler and realod values.
+ *
+ */
+IWDG_t::IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload) {
+  // assert(IS_IWDG_PRESCALER(iwdg_prescaler)) Should this be added since it is needed when reinitialzing the watchdog
+  // assert(IS_IWDG_RELOAD(iwdg_reload))
+	this->iwdg_prescaler = iwdg_prescaler;
+	this->iwdg_reload = iwdg_reload;
+}
+
+/**
+ * @brief Reset the reload and counter values inside a watchdog
+ * using the member variables of IWDG_t
+ *
+ */
+HAL_StatusTypeDef IWDG_t::InitWatchdog(IWDG_HandleTypeDef *watchdog) {
+	watchdog->Init.Reload = this->iwdg_reload;
+	watchdog->Init.Prescaler = this->iwdg_prescaler;
+	HAL_StatusTypeDef refresh = HAL_IWDG_Init(watchdog);
+	return refresh;
+}
+
+/**
+ * @brief Refresh the watchdog in the parameter. It will reset the downcounter
+ * to the value within the reload register.
+ *
+ */
+HAL_StatusTypeDef IWDG_t::RefreshWatchdog(IWDG_HandleTypeDef *watchdog) {
```

Comment:
return a bool rather than HAL_StatusTypeDef here. perhaps return false for an error or something like that? Or make a status bool stored in the driver that can be checked by a manager. @TheSpaceDragon @Chrisc110 @Ayounggie  Thoughts?

Reply:
Yeah, I agree with changing the return type to bool. It would be nice to have something like didWatchdogRefresh confirmation after the function returns.

Reply:
Should I use try catch to supress errors and return false as well in the newest commit?